tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -514,21 +487,19 @@ void StorageManagerCanonical::write_consolidated_commits_file(
   // Write the file to storage.
   URI consolidated_commits_uri =
       array_dir.get_commits_dir(write_version)
-          .join_path(name.value() + constants::con_commits_file_suffix);
+          .join_path(name + constants::con_commits_file_suffix);
   throw_if_not_ok(
       vfs()->write(consolidated_commits_uri, data.data(), data.size()));
   throw_if_not_ok(vfs()->close_file(consolidated_commits_uri));
 }
 
 void StorageManagerCanonical::delete_array(const char* array_name) {
   if (array_name == nullptr) {
-    throw Status_StorageManagerError(
-        "[delete_array] Array name cannot be null");
+    throw std::invalid_argument("[delete_array] Array name cannot be null");
   }
 
   // Delete fragments and commits
-  throw_if_not_ok(
-      delete_fragments(array_name, 0, std::numeric_limits<uint64_t>::max()));
+  delete_fragments(array_name, 0, std::numeric_limits<uint64_t>::max());
```

Comment:
`delete_fragments` is loading `ArrayDirectory` from disk to perform its operations, and the next line in this function does exactly the same thing. Loading `ArrayDirectory` is an expensive operation and we should not be doing that twice without an important reason.
I realize this is not part of this PR, so if we don't want to address it here we should open a followup ticket.

Reply:
Both `delete_fragments` and `delete_array` will be moved _into_ the `ArrayDirectory` (as `delete_fragments_list`) in a future work item, which _should_ resolve this issue. I've made a note to ensure that's an integrated change.